### ISSUE TITLE
[1.6] - meteor npm i - Cannot read property '0' of undefined

### ISSUE THREAD

**Issue by zeroasterisk** [meteor/meteor#9333]
Bug:

I had this when testing `rc` for 1.6 (mentioned in the PR thread)

I am now testing the actual 1.6 release, and I'm getting the same thing:

`Cannot read property '0' of undefined`

This is very similar to #9319 but not exactly the same, not related to lodash/underscore.

npm debug log for: `meteor npm i`
http://sebsauvage.net/paste/?ed554c3cb23a93d8#HAsff8J+quOmXQNqnKdARP+leXFhCcZmNO1fdsUSOxI=

npm debug log for: `meteor npm i react-rangeslider --save`
http://sebsauvage.net/paste/?127b43824386dd16#8tPHLT4cSWYYyqelzETfsQwBJnv2LuLAzkS4Z72ENQU=

```
$ meteor --version && meteor npm --version && uname -a
Meteor 1.6
5.4.2
Darwin alan 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEASE_X86_64 x86_64 i386 MacBookPro11,5 Darwin
```

NOTE:

If I trash all meteor linker files and node_modules, I can then run npm commands again

```
$ rm -rf .meteor/local/bundler-cache/linker && rm -rf node_modules
$ meteor npm i react-rangeslider --save

> dtrace-provider@0.8.5 install /Users/alan/Code/eltoro/portalV2/webapp/node_modules/bunyan/node_modules/dtrace-provider
> node scripts/install.js


> fsevents@1.1.2 install /Users/alan/Code/eltoro/portalV2/webapp/node_modules/fsevents
> node install

[fsevents] Success: "/Users/alan/Code/eltoro/portalV2/webapp/node_modules/fsevents/lib/binding/Release/node-v57-darwin-x64/fse.node" already installed
Pass --update-binary to reinstall or --build-from-source to recompile

> bcrypt@1.0.3 install /Users/alan/Code/eltoro/portalV2/webapp/node_modules/bcrypt
> node-pre-gyp install --fallback-to-build

[bcrypt] Success: "/Users/alan/Code/eltoro/portalV2/webapp/node_modules/bcrypt/lib/binding/bcrypt_lib.node" is installed via remote

> node-sass@4.5.3 install /Users/alan/Code/eltoro/portalV2/webapp/node_modules/node-sass
> node scripts/install.js

Cached binary found at /Users/alan/.npm/node-sass/4.5.3/darwin-x64-57_binding.node

> uglifyjs-webpack-plugin@0.4.6 postinstall /Users/alan/Code/eltoro/portalV2/webapp/node_modules/uglifyjs-webpack-plugin
> node lib/post_install.js


> node-sass@4.5.3 postinstall /Users/alan/Code/eltoro/portalV2/webapp/node_modules/node-sass
> node scripts/build.js

Binary found at /Users/alan/Code/eltoro/portalV2/webapp/node_modules/node-sass/vendor/darwin-x64-57/binding.node
Testing binary
Binary is fine
+ react-rangeslider@2.2.0
added 2019 packages in 37.763s
```

but it errors out again, immediately after:

```
$ meteor npm i
npm ERR! Cannot read property '0' of undefined

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/alan/.npm/_logs/2017-11-09T16_22_14_531Z-debug.log
```

**Comment by zeroasterisk** [comment]
NOTE: I have tried to create a reproduction [repo here](https://github.com/zeroasterisk/meteor-issue-9333)
```
$ meteor create meteor-issue-9333
$ cd meteor-issue-9333
$ cp <path>.meteor/packages .meteor/
$ cp <path>.package.json ./
$ meteor npm i
$ meteor
...
$ meteor npm i
up to date in 6.38s
$ meteor npm i react-rangeslider --save
+ react-rangeslider@2.2.0
added 2 packages in 7.91s
```

**Comment by QHose** [comment]
hi, delete your `node modules` dir and run meteor again.

**Comment by yanickrochon** [comment]
This issue is discussed [here](https://github.com/npm/npm/issues/19068), [here](https://github.com/npm/npm/issues/18238), [here](https://github.com/npm/npm/issues/18042), [here](https://github.com/npm/npm/issues/17858), etc. :stuck_out_tongue:  Basically, it's an npm issue, not Meteor.

**Comment by zeroasterisk** [comment]
@QHose I did that several times, but I never included the `--save` on the blank `npm install` command.

So I tried it again
`rm -rf node_modules && meteor npm install --save` 

Same error:

```
npm ERR! Cannot read property '0' of undefined

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/alan/.npm/_logs/2017-11-09T17_52_02_979Z-debug.log
```

**Comment by peernohell** [comment]
@zeroasterisk In addition to remove the node_modules can you try to remove the package-lock.json 
 and then make a `meteor npm install`?
it's solved the issue for me

**Comment by hwillson** [comment]
Looks like this issue has been fixed upstream (https://github.com/npm/npm/issues/17858#issuecomment-344522435) and is coming in the next npm release.

**Comment by hwillson** [comment]
This was fixed in npm 5.6.0, which is included with Meteor 1.6.1. Closing - thanks!
